### DISCUSSION TITLE
How to get UI updates after mutation that creates a new object

### DISCUSSION THREAD

**Discussion by sheldonbaker** [urql-graphql/urql#2127] [Q&A] [answered]
I am having a problem with a fairly basic use case (using the Vue bindings). I create a Widget via a mutation and then display the newly-created widget in my template. But if that widget has a property updated by another mutation thereafter, the widget's data in the template is not updated, even though the cache has picked up the newly-updated title value.

```
setup() {
  const { data, executeMutation } = useMutation(CREATE_WIDGET)
  return { data, executeMutation }
}
```
```
<template>
{{ data.createWidget.title }} - this `title` attribute does not get updated after a hypothetical separate `UPDATE_WIDGET_TITLE` mutation
</template>
```

Is this expected behaviour? Do I need to need to do something like `useQuery(GET_WIDGET)` once the `CREATE_WIDGET` mutation is completed and use that in my template instead?

**Comment by JoviDeCroock** [comment, marked as answer]
Yes, mutations don't connect to the cache. Queries are used to keep a continuous observation of cached data
